src/analysis/asymptotics/asymptotics.lean

```diff
@@ -1170,29 +1170,53 @@ end sum
 
 /-! ### Relation between `f = o(g)` and `f / g → 0` -/
 
-theorem is_o.tendsto_0 {f g : α → 𝕜} {l : filter α} (h : is_o f g l) :
+theorem is_o.tendsto_div_nhds_zero {f g : α → 𝕜} {l : filter α} (h : is_o f g l) :
   tendsto (λ x, f x / (g x)) l (𝓝 0) :=
 have eq₁ : is_o (λ x, f x / g x) (λ x, g x / g x) l,
   by simpa only [div_eq_mul_inv] using h.mul_is_O (is_O_refl _ _),
 have eq₂ : is_O (λ x, g x / g x) (λ x, (1 : 𝕜)) l,
-  from is_O_of_le _ (λ x, by by_cases h : ∥g x∥ = 0; simp [h, zero_le_one]),
+  from is_O_of_le _ (λ x, by simp [div_self_le_one]),
 (is_o_one_iff 𝕜).mp (eq₁.trans_is_O eq₂)
 
+theorem is_o.tendsto_inv_smul_nhds_zero [normed_space 𝕜 E'] {f : α → E'} {g : α → 𝕜} {l : filter α}
+  (h : is_o f g l) : tendsto (λ x, (g x)⁻¹ • f x) l (𝓝 0) :=
+by simpa only [div_eq_inv_mul, ← normed_field.norm_inv, ← norm_smul,
+  ← tendsto_zero_iff_norm_tendsto_zero] using h.norm_norm.tendsto_div_nhds_zero
+
 theorem is_o_iff_tendsto' {f g : α → 𝕜} {l : filter α}
     (hgf : ∀ᶠ x in l, g x = 0 → f x = 0) :
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
-iff.intro is_o.tendsto_0 $ λ h,
+iff.intro is_o.tendsto_div_nhds_zero $ λ h,
   (((is_o_one_iff _).mpr h).mul_is_O (is_O_refl g l)).congr'
     (hgf.mono $ λ x, div_mul_cancel_of_imp) (eventually_of_forall $ λ x, one_mul _)
 
 theorem is_o_iff_tendsto {f g : α → 𝕜} {l : filter α}
     (hgf : ∀ x, g x = 0 → f x = 0) :
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
-⟨λ h, h.tendsto_0, (is_o_iff_tendsto' (eventually_of_forall hgf)).2⟩
+⟨λ h, h.tendsto_div_nhds_zero, (is_o_iff_tendsto' (eventually_of_forall hgf)).2⟩
 
 alias is_o_iff_tendsto' ↔ _ asymptotics.is_o_of_tendsto'
 alias is_o_iff_tendsto ↔ _ asymptotics.is_o_of_tendsto
 
+@[simp] lemma is_o_const_left {c : E'} :
+  is_o (λ x, c) g' l ↔ c = 0 ∨ tendsto (norm ∘ g') l at_top :=
```

Comment:
Is `(norm ∘ g')` preferred over a lambda expression? It looks to me like switching to a lambda would prevent things like `simp only [(∘)]` in later files, but on the other hand pointfree style is nice! Also does `g` work here instead of `g'`?

Reply:
I don't think that we have a fixed coding style here. I slightly prefer `∘` because I think that it is a bit more readable.